tests/utils/test_sampling.py

```
@@ -1197,6 +1197,444 @@ def check_result(result, valid):
+def _reference_hy3_fused_sampler_candidates(
+    logits,
+    penalty_mask,
+    slot_id,
+    repetition_penalty,
+    temperature,
+    softmax_policy,
+    top_k,
+    max_top_k,
+):
+    """Build the sorted candidate state without calling a FlashInfer sampler."""
+    batch_size, vocab_size = logits.shape
```

Ruff (0.16.2)

[warning] Unpacked variable `batch_size` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)